packages/core/src/modules/credentials/formats/indy/IndyCredentialFormatService.ts

```diff
@@ -0,0 +1,750 @@
+   * @returns The created credential offer
+   */
+  public async createOffer(
+    proposal: AcceptProposalOptions | NegotiateProposalOptions | OfferCredentialOptions
+  ): Promise<CredProposeOfferRequestFormat> {
+    if (this.indyIssuerService && proposal.credentialFormats?.indy?.credentialDefinitionId) {
+      const credOffer: CredOffer = await this.indyIssuerService.createCredentialOffer(
+        proposal.credentialFormats.indy.credentialDefinitionId
+      )
+      return {
+        indy: {
+          payload: {
+            credentialPayload: credOffer, // old v1 object from Indy SDK
+          },
+        },
+      }
+    }
+    if (!this.indyIssuerService) {
+      throw new Error('Missing Indy Issuer Service')
+    } else {
+      throw new Error('Missing Credential Definition id')
+    }
+  }
+
+  /**
+   * Create a credential offer for the given credential definition id.
+   *
+   * @param options RequestCredentialOptions the config options for the credential request
+   * @throws Error if unable to create the request
+   * @returns The created credential offer
+   */
+  private async createRequest(options: RequestCredentialOptions): Promise<CredProposeOfferRequestFormat> {
+    if (
+      this.indyHolderService &&
+      options.holderDid &&
+      options.offer &&
+      options.offer.indy?.payload.credentialPayload &&
+      options.credentialDefinition &&
+      options.credentialDefinition.indy?.credDef
+    ) {
+      const credoffer: CredOffer = options.offer.indy?.payload.credentialPayload as CredOffer
+      const [credReq, credReqMetadata] = await this.indyHolderService.createCredentialRequest({
+        holderDid: options.holderDid,
+        credentialOffer: credoffer,
+        credentialDefinition: options.credentialDefinition.indy?.credDef,
+      })
+      const request: CredProposeOfferRequestFormat = {
+        indy: {
+          payload: {
+            credentialPayload: credReq,
+            requestMetaData: credReqMetadata,
+          },
+        },
+      }
+      return request
+    }
+    throw Error('Unable to create Credential Request')
+  }
+
+  /**
+   * Method to insert a preview object into a proposal. This can occur when we retrieve a
+   * preview object as part of the stored credential record and need to add it to the
+   * proposal object used for processing credential proposals
+   * @param proposal the proposal object needed for acceptance processing
+   * @param preview the preview containing stored attributes
+   * @returns proposal object with extra preview attached
+   */
+  public setPreview(proposal: AcceptProposalOptions, preview: V2CredentialPreview): AcceptProposalOptions {
+    if (proposal.credentialFormats.indy) {
+      proposal.credentialFormats.indy.attributes = preview.attributes
+    }
+    return proposal
+  }
+
+  public async processProposal(
+    options: AcceptProposalOptions,
+    credentialRecord: CredentialExchangeRecord
+  ): Promise<AcceptProposalOptions> {
```

Comment:
Why are we already returning the accept options here? I think we can maybe leave that to the create request method